### PR TITLE
libraries: SPI: Hanlde multiple SPI instances

### PR DESCRIPTION
**Pull request by soburi**
Create multiple SPI instance if spis array contains plural elements. Declare these as 'SPI1', 'SPI2', ..., from second element of the array. (First element is already declare as 'SPI'.)

### CODE REVIEW

File: libraries/SPI/SPI.cpp [hunk 1]
```diff
@@ -121,5 +121,28 @@ void arduino::ZephyrSPI::begin() {}
...
+
+#undef DECLARE_SPI_N
+#undef DECL_SPI_N
+#undef DECL_SPI_0
+#undef ARDUINO_SPI_DEFINED_0
+#else  // PROP_LEN(spis) > 1
+/* When PROP_LEN(spis) == 1, DT_FOREACH_PROP_ELEM work not correctly. */
 arduino::ZephyrSPI
     SPI(DEVICE_DT_GET(DT_PHANDLE_BY_IDX(DT_PATH(zephyr_user), spis, 0)));
+#endif // HAS_PORP(i2cs)
```
Comment (DhruvaG2000) [inline libraries/SPI/SPI.cpp]: Pardon me, I didn't understand how I2C has any relevance here in SPI.cpp?

File: libraries/SPI/SPI.cpp [hunk 2]
```diff
@@ -121,5 +121,28 @@ void arduino::ZephyrSPI::begin() {}
...
+/* If i2cs node is not defined, tries to use arduino_i2c */
```
Comment (DhruvaG2000) [inline libraries/SPI/SPI.cpp]: `s/arduino_i2c/arduino_spi` ?

**Review by DhruvaG2000 — CHANGES_REQUESTED** [review]

File: libraries/SPI/SPI.cpp (hunk 1 above)
Comment (soburi) [reply libraries/SPI/SPI.cpp]: Sorry, I verified the code but forgot to edit the comment. I'm using the structure that was originally used for I2C.

File: libraries/SPI/SPI.cpp (hunk 2 above)
Comment (soburi) [reply libraries/SPI/SPI.cpp]: Fixed it. Thanks.

**Comment by soburi** [comment]
I'm going to put it back into draft because I have some concerns. Sorry.

**Review by DhruvaG2000 — COMMENTED** [review]
@soburi  nothing against this PR, but can you add more in your commit about how this change helps us/ what feature it enables? Talk about what enhancement this is bringing in basically.

**Comment by DhruvaG2000** [comment]
Also, `s/Hanlde/handle` in $subject

File: libraries/SPI/SPI.cpp [hunk 3]
```diff
@@ -121,5 +121,28 @@ void arduino::ZephyrSPI::begin() {}
...
+#undef DECL_SPI_N
+#undef DECL_SPI_0
+#undef ARDUINO_SPI_DEFINED_0
+#else  // PROP_LEN(spis) > 1
+/* When PROP_LEN(spis) == 1, DT_FOREACH_PROP_ELEM work not correctly. */
 arduino::ZephyrSPI
     SPI(DEVICE_DT_GET(DT_PHANDLE_BY_IDX(DT_PATH(zephyr_user), spis, 0)));
+#endif // HAS_PORP(spis)
+/* If spis node is not defined, tries to use arduino_spi */
+#elif DT_NODE_EXISTS(DT_NODELABEL(arduino_spi))
```
Comment (Ayush1325) [inline libraries/SPI/SPI.cpp:144]: So where does the `arduino-spi` name come from? Are any boards already defining it using that label?

**Review by Ayush1325 — COMMENTED** [review]
LGTM

File: libraries/SPI/SPI.cpp (hunk 3 above)
Comment (soburi) [reply libraries/SPI/SPI.cpp:144]: It is usually define with `arduino_header` in `zephyr/boards` definitions.
This line is intended to take advantage of it if it is already defined.
And we can override with overlay in this package with defining `spis` property.

**Comment by soburi** [comment]
@DhruvaG2000 

> @soburi nothing against this PR, but can you add more in your commit about how this change helps us/ what feature it enables? Talk about what enhancement this is bringing in basically.

Added description to commit message. 

> Also, s/Hanlde/handle in $subject

Fixed it. Thank you.

**Comment by DhruvaG2000** [comment]
> > Also, s/Hanlde/handle in $subject
> 
> Fixed it. Thank you.

@soburi  Nope, I still see the wrong spelling in subject. https://github.com/zephyrproject-rtos/gsoc-2022-arduino-core/pull/113/commits

File: libraries/SPI/SPI.h [hunk 4]
```diff
@@ -56,7 +56,24 @@ class ZephyrSPI : public HardwareSPI {
 
 } // namespace arduino
 
+#if DT_NODE_HAS_PROP(DT_PATH(zephyr_user), spis) && (DT_PROP_LEN(DT_PATH(zephyr_user), spis) > 1)
+#define ARDUINO_SPI_DEFINED_0 1
+#define DECL_EXTERN_SPI_0(i)  extern arduino::ZephyrSPI SPI;
+#define DECL_EXTERN_SPI_N(i)  extern arduino::ZephyrSPI SPI##i;
+#define DECLARE_EXTERN_SPI_N(n, p, i)                                                             \
+        COND_CODE_1(ARDUINO_SPI_DEFINED_##i, (DECL_EXTERN_SPI_0(i)), (DECL_EXTERN_SPI_N(i)))
+
```
Comment (DhruvaG2000) [inline libraries/SPI/SPI.h:65]: Please fix above checkpatch warns @soburi .

>  WARNING: please, no spaces at the start of a line

**Review by DhruvaG2000 — CHANGES_REQUESTED** [review]

File: libraries/SPI/SPI.h (hunk 4 above)
Comment (DhruvaG2000) [reply libraries/SPI/SPI.h:65]: Was this fixed?
Can you please fix, rebase and force push if not done?

**Review by DhruvaG2000 — CHANGES_REQUESTED** [review]

**Review by DhruvaG2000 — APPROVED** [review]

**Pull request merged**